content_handler/service_protocol_hooks.cc

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/service_protocol_hooks.h"
+
+#include <string.h>
+
+#include <string>
+#include <vector>
+
+#include "flutter/common/threads.h"
+#include "flutter/content_handler/app.h"
+#include "lib/ftl/memory/weak_ptr.h"
```

Comment:
My understanding is that the content handler and shell are separate entry points into Flutter based on different stacks, so the content handler can't use code from the shell.